docs/Developers/Git_Basics.rst

```diff
@@ -283,6 +283,35 @@ another editor will be fired up with all the messages of commits which you want
 to squash. Clean up all the messages and add a new message to be
 displayed for the single commit.
 
+Common Git Issues
+-----------------
+Many a times , it might happen that after rebasing, you use ``git add -A`` which
+in turn commit those files into your branch and push it to the remote , you
+actually don't want to. Here ,is a short outline of, how can you remove (or
+revert changes in) particular files from your commit even after pushing it to
+the remote .
+
+In your local repo , to revert the file to the state before the last commit run:
+::
+
+    $ git checkout HEAD^ /path/to/file
+
+Now , after reverting the file(s) update your last commit , by running :
+::
+
+    $ git commit --amend
+
+To apply these changes to the remote you need to force update the branch :
+::
+
+    $ git push -f myfork
+
+Besides, the ``git checkout`` command exhibits more flexibility in reverting the
+changes in a file, done even before the last commit , by replacing the ``HEAD^``
+in the above command by the revision number of the particular HEAD commit, you
+want to refer to.
+
```

Comment:
Maybe add an example. This is for newcomers so they might not know what exactly to do here.